airflow/hooks/fs_hooks/s3.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
```

Comment:
This shouldn't be a new hook - it should be methods added to the existing S3Hook.

Reply:
I agree, otherwise these will diverge and people will get confused

Reply:
I don't necessarily agree, as the current S3 hook represents S3 more as a key-based blob store than a file system. See my comment below for more details. A compromise may be to combine the two interfaces, allowing the hook to be used as a file system or the current interface. This would bloat the interface somewhat though.